client/dashboard/data/user-tax-details.ts

```diff
@@ -0,0 +1,92 @@
+import wpcom from 'calypso/lib/wp';
+import type { Field, NormalizedField } from '@wordpress/dataviews';
+
+export interface UserTaxFormData {
+	country: string;
+	id: string;
+	address: string;
+	name: string;
+}
+
+export interface UserTaxDetails {
+	country?: string | null;
+	id?: string | null;
+	address?: string | null;
+	name?: string | null;
+	isForBusiness?: boolean | null;
+	can_user_edit?: boolean | false;
+}
+
+export type UserTaxField = Field< UserTaxFormData > & {
+	isDisabled?: boolean;
+	isVatAlreadySet?: boolean;
+	canUserEdit?: boolean;
+	taxName?: string;
+};
+
+export type UserTaxNormalizedField = NormalizedField< UserTaxFormData > & {
+	isDisabled?: boolean;
+	isVatAlreadySet?: boolean;
+	canUserEdit?: boolean;
+	taxName?: string;
+};
+
+export interface UserTaxFormControlProps {
+	data: UserTaxFormData;
+	field: UserTaxNormalizedField;
+	onChange: ( edits: Partial< UserTaxFormData > ) => void;
+}
+
+export async function fetchUserTaxDetails(): Promise< UserTaxFormData > {
+	return await wpcom.req.get( '/me/vat-info' );
+}
+
+export async function updateUserTaxDetails(
+	data: Partial< UserTaxFormData >
+): Promise< Partial< UserTaxFormData > > {
+	const savableKeys = [ 'country', 'id', 'name', 'address' ];
+	for ( const key in data ) {
+		if ( ! savableKeys.includes( key ) ) {
+			delete data[ key as keyof UserTaxFormData ];
+		}
+	}
+	return await wpcom.req.post( '/me/vat-info', data );
+}
+
+export interface CountryListItemBase {
```

Comment:
Is this basically the same type as https://github.com/Automattic/wp-calypso/pull/105075/files#diff-5c981ee71dce0b0f4c57bb191ece2a2c605050314ae0ebfe60c69cdf7a47cfdcR3? cc: @paulopmt1 

Maybe we should extract this to `data/supported-countries.ts` instead (maybe later as follow-up).